frontend/src/components/modals/settings/SettingsModal.tsx

```diff
@@ -92,24 +109,25 @@ function SettingsModal({ isOpen, onOpenChange }: SettingsProps) {
     );
   };
 
-  const isDisabled =
```

Comment:
Removing this constant prevents two behaviors:
- Disable save if one of the inputs are empty
- Disable save if there are no changed values

You can find those tests failed. If this was intentional, ignore this comment, but be cautioned that the user can force reinitialize the agent by pressing "Save" (with no changes)

I will adjust the other failing tests according to the latest changes of settings and disable behavior once this is merged.

Reply:
Hmm good point. Part of this was intentional--you do need to re-save if there's a migration, even if no settings are changed

Probably good to prevent certain values from being empty though